pkg/auth/cleanup/service_test.go

```diff
@@ -141,93 +159,108 @@ func newMockCleanupService(t *testing.T,
 	grbStore map[string]*v3.GlobalRoleBinding,
 	prtbStore map[string]*v3.ProjectRoleTemplateBinding,
 	crtbStore map[string]*v3.ClusterRoleTemplateBinding,
+	tokenStore map[string]*v3.Token,
 	userStore map[string]*v3.User,
 	secretStore map[string]*v1.Secret) Service {
 	t.Helper()
 	ctrl := gomock.NewController(t)
-	grbCache := fake.NewMockNonNamespacedCacheInterface[*v3.GlobalRoleBinding](ctrl)
-	grbCache.EXPECT().List(gomock.Any()).DoAndReturn(func(_ labels.Selector) ([]*v3.GlobalRoleBinding, error) {
-		var lst []*v3.GlobalRoleBinding
-		for _, v := range grbStore {
-			lst = append(lst, v)
-		}
-		return lst, nil
-	}).AnyTimes()
-	grbCache.EXPECT().Get(gomock.Any()).DoAndReturn(func(name string) (*v3.GlobalRoleBinding, error) {
-		return grbStore[name], nil
-	}).AnyTimes()
-	grbClient := fake.NewMockNonNamespacedClientInterface[*v3.GlobalRoleBinding, *v3.GlobalRoleBindingList](ctrl)
-	grbClient.EXPECT().Delete(gomock.Any(), gomock.Any()).DoAndReturn(func(name string, _ *metav1.DeleteOptions) error {
-		delete(grbStore, name)
-		return nil
-	}).AnyTimes()
 
-	prtbCache := fake.NewMockCacheInterface[*v3.ProjectRoleTemplateBinding](ctrl)
-	prtbCache.EXPECT().List(gomock.Any(), gomock.Any()).DoAndReturn(func(_ string, _ labels.Selector) ([]*v3.ProjectRoleTemplateBinding, error) {
-		var lst []*v3.ProjectRoleTemplateBinding
-		for _, v := range prtbStore {
+	grbCache := initMockCache(ctrl, grbStore)
```

Comment:
IMHO extracting mocks to `initX` functions was detrimental as it didn't help with neither readability nor with ability to set exact expectations.

Reply:
Yeah, it started little by little. They seemed to have the same expectations, and then one after one I realized that I needed to add some differences, and so they became a bit useless.

Reverted with 08fbfdb.

The setup of the mock is still pretty long, but that's ok.